community-containers/npmplus/readme.md

```diff
@@ -0,0 +1,17 @@
+## NPMplus
+This container contains a fork of the Nginx Proxy Manager, which is a WebUI for nginx. It will also automatically create a config and cert for AIO.
+
+### Notes
+- Make sure that no other service is using port `80 (tcp)`, `443 (tcp/upd)` or `81 (udp)` on your host as otherwise the containers will fail to start. You can check this with `sudo netstat -tulpn | grep "443\|80\|81"` before installing AIO.
```

Comment:
I guess this will conflict with the default AIO instructions... Maybe there should be an additional hint that the command for AIO itself needs to be adjusted? I suppose there is no way for NPM to retreive certs via the tls-alpn challenge? This would eliminate this problem :)

Reply:
thats why I asked https://github.com/nextcloud/all-in-one/discussions/3505, I used the caddy community container as base to create this community container and there is also no mention of this

Reply:
port 80 of mastercontainer always redirects to https, right?

Reply:
yes but it is also used for getting the cert for the AIO interface... So if port 80 is removed from there then only port 8080 can be used...

Reply:
then I just add DISABLE_HTTP=true and it should work, the users are then forced to use https for all sites

Reply:
So NPM can actually get certificates via tls-alpn challenge? :)

Reply:
no, but the http challenge support https, if http redirects to the same domain on https, it ignores self signed or invalid certs on https

Reply:
interesting, I didnt know! So I guess this should actually make it work then? :)
https://github.com/nextcloud/all-in-one/blob/5dcdb6268ab2b08b58f7f2510de3bfe01bc55bbc/Containers/mastercontainer/Caddyfile#L23-L25

Reply:
yes

Reply:
https://github.com/ZoeyVid/NPMplus/blob/920e4848fa731563dec9535c8cbbb93626af57ad/Caddyfile#L28 I know with this line it works

Reply:
So maybe permanend should be added?

Reply:
https://github.com/nextcloud/all-in-one/pull/3522